### PR TITLE
feat: update timelock unlock condition

### LINKED ISSUE #7983: [Task]: update timelock unlock condition logic
### Task description

Update the logic to calculate if the time has expired based on the `slot index` and `min_committable_age`, instead of calculating it with the unix timestamp

https://github.com/iotaledger/iota-sdk/blob/2.0/sdk/src/types/block/output/unlock_condition/timelock.rs#L34

### PR DESCRIPTION
**Pull request by cpl121**
Closes #7983 

## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

...

## Changelog

```
- Please write a list of changes
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/wallet/utils/outputs/isOutputTimeLocked.ts [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
-    return unlockTime !== undefined && unlockTime.getTime() > Date.now()
+    const output = outputData.output as CommonOutput
+    const timelockUnlockCondition = output.unlockConditions.find(
+        (unlockCondition) => unlockCondition.type === UnlockConditionType.Timelock
+    ) as TimelockUnlockCondition
+    const nodeProtocolParameters = get(nodeInfoProtocolParameters)
+    if (!nodeProtocolParameters || !timelockUnlockCondition) {
+        return true
+    } else {
+        return timelockUnlockCondition.slotIndex + nodeProtocolParameters.minCommittableAge < 0
```
Comment (marc2332) [inline packages/shared/lib/core/wallet/utils/outputs/isOutputTimeLocked.ts:14]: I don't think this makes sense, this is not the same as the rust [code](https://github.com/iotaledger/iota-sdk/blob/2.0/sdk/src/types/block/output/unlock_condition/timelock.rs#L34).

The correct (I Think) formula is: `(currentSlotIndex + minCommittableAge) < output.timelockUnlockCondition.slotIndex`

but you are doing: `(output.timelockUnlockCondition.slotIndex + + minCommittableAge) < 0`

If you think about it, you can't check if an output is locked if you don't even know at what moment you are when you check it.

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: packages/shared/lib/core/wallet/utils/outputs/isOutputTimeLocked.ts (hunk 1 above)
Comment (marc2332) [reply packages/shared/lib/core/wallet/utils/outputs/isOutputTimeLocked.ts:14]: From another POV, there is no way that your `(output.timelockUnlockCondition.slotIndex + minCommittableAge)` can ever be `< 0`

Comment (marc2332) [reply packages/shared/lib/core/wallet/utils/outputs/isOutputTimeLocked.ts:14]: So I think the correct code would be something like 

```ts
(currentSlotIndex + nodeProtocolParameters.minCommittableAge) < timelockUnlockCondition.slotIndex
```

I don't know where you can't get `currentSlotIndex` from tho

Comment (marc2332) [reply packages/shared/lib/core/wallet/utils/outputs/isOutputTimeLocked.ts:14]: Ah, I see there is `getSlotIndexFromNodeInfo` you can use that for `currentSlotIndex`

**Review by marc2332 — APPROVED** [review]
:godmode: :zap:  I approve this :+1: :deciduous_tree:

**Review by begonaalvarezd — APPROVED** [review]
🔥

**Pull request merged**